src/inmanta/execute/proxy.py

```diff
@@ -62,7 +62,28 @@ class ProxyMode(enum.Enum):
     EXPORT = enum.auto()
 
 
-@dataclasses.dataclass(kw_only=True, frozen=True, slots=True)
+global_proxy_mode: ProxyMode = ProxyMode.PLUGIN
```

Comment:
I think it's an improvement to pull this out of the context object. I do wonder if `contextvars` might be more appropriate than a global. I have to confess I've never used them before, but it seems like a perfect fit to me. And I'd expect it to be performant, but not sure.

Reply:
Ordinarily: yes. 

Will it be performant: yes.
Will it be as performant as this: no.

Is this bad: I doubt it. It prevent running the compiler in parallel with itself. Which is nasty. But I don't think it is the only thing. 

I can probably measure the difference

Reply:
Sees to be within the noise margin for the test I have, I'll use context vars